File: src/types/project.ts
export interface Project {
  id: number;
  slug: string;
  titleEn: string;
  titleAr: string;
  client_id?: number | null;
  client_name?: string | null;
  descriptionEn?: string | null;
  descriptionAr?: string | null;
  challenge_en?: string | null;
  challenge_ar?: string | null;
  solution_en?: string | null;
  solution_ar?: string | null;
  key_features_en?: string | null; // JSON array string
  key_features_ar?: string | null; // JSON array string
  results_achieved_en?: string | null; // JSON array string
  results_achieved_ar?: string | null; // JSON array string
  technologiesEn?: string | null; // JSON array string
  technologiesAr?: string | null; // JSON array string
  project_date?: string | null;
  category?: string | null;
  featured?: number | boolean;
  imageUrl?: string | null;
  gallery_images?: string | null; // JSON encoded array from API
  demoUrl?: string | null;
  codeUrl?: string | null;
  accent_color?: string | null;
  status?: string | null;
  order?: number | null;
  createdAt?: string | null;
  updatedAt?: string | null;
}

export interface ProjectListItem {
  id: number;
  slug: string;
  titleEn: string;
  titleAr: string;
  client_id?: number | null;
  client_name?: string | null;
  project_date?: string | null;
  category?: string | null;
  featured?: number | boolean;
  imageUrl?: string | null;
  demoUrl?: string | null;
  codeUrl?: string | null;
}

export interface ProjectApiResponse {
  status: 'success' | 'failure';
  data?: Project[] | Project;
  message?: string;
}

export interface ProjectsApiResponse {
  status: 'success' | 'failure';
  data?: ProjectListItem[];
  message?: string;
}

File: backend/BusinessLogiclayer/Entities/Project.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BusinessLogiclayer.Entities
{
    public class Project
    {
        public int Id { get; set; }
        public string Slug { get; set; } = string.Empty;
        public string TitleEn { get; set; } = string.Empty;
        public string TitleAr { get; set; } = string.Empty;
        public string ClientName { get; set; } = string.Empty;
        public DateTime ProjectDate { get; set; } 
        public ProjectCategory Category { get; set; } = ProjectCategory.WebDevelopment;
        public string ImageUrl { get; set; } = string.Empty;
        public string DemoUrl { get; set; } = string.Empty;
    }
}
